core/arch/arm/mm/tee_pager.c

```diff
@@ -541,6 +541,30 @@ static bool pager_check_access(struct abort_info *ai)
 
 }
 
+#ifdef CFG_TEE_CORE_DEBUG
+static void stat_handle_fault(void)
+{
+	static size_t num_faults;
+	static size_t min_npages = SIZE_MAX;
+	static size_t total_min_npages = SIZE_MAX;
+
+	num_faults++;
+	if ((num_faults % 1024) == 0 || tee_pager_npages < total_min_npages) {
+		DMSG("nfaults %zu npages %zu (min %zu)",
+		     num_faults, tee_pager_npages, min_npages);
+		min_npages = tee_pager_npages; /* reset */
+	}
+	if (tee_pager_npages < min_npages)
+		min_npages = tee_pager_npages;
+	if (tee_pager_npages < total_min_npages)
+		total_min_npages = tee_pager_npages;
+}
+#else
+static void stat_handle_fault(void)
```

Comment:
As far as I know that wouldn't make any difference. It may or may not be inlined regardless of `inline`.